### PR TITLE
feat: langchain update

### PR DESCRIPTION
**Pull request by louisgv**
based on top of #1 

- resolves https://linear.app/openrouter/issue/OPE-82/update-the-langchain-stuff-to-use-openrouter

### CODE REVIEW

**Review by alexanderatallah — COMMENTED** [review]
added some comments on #1 that apply here too

**Comment by louisgv** [comment]
@alexanderatallah resolved the constants and default model issue - regarding the return result hack, maybe a disparity between openai's client API and openrouter? We would need to dig into openai's client code and see which type of result get transformed into JSON. I suspect the openai client might be concating the stream, and if the first result is valid json, it returns JSON, otherwise it returns plain text.

File: shared/__pycache__/constants.cpython-310.pyc
Comment (alexanderatallah) [inline shared/__pycache__/constants.cpython-310.pyc:1]: should be gitignored

File: shared/constants.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+OPENROUTER_REFERER = "https://openrouter.streamlit.io"
```
Comment (alexanderatallah) [inline shared/constants.py:1]: nit: correct spelling is referrer (except when used as a header)

File: components/__pycache__/Sidebar.cpython-310.pyc
Comment (alexanderatallah) [inline components/__pycache__/Sidebar.cpython-310.pyc:1]: should be gitignored

**Pull request merged**